StandardSocketsHttpHandler/Net/Http/Extensions/HttpRequestMessageExtensions.cs

```diff
@@ -6,20 +6,15 @@ namespace System.Net.Http
 {
     internal static class HttpRequestMessageExtensions
     {
+        // Depending on runtime the backing field for headers is either _headers or headers:
+        // _headers in .net core and some .net fx and mono versions (.NET Framework 4.6.1, Mono on Mac)
+        // headers in other .net fx / mono versions.
+        private static readonly FieldInfo s_headersField = typeof(HttpRequestMessage).GetField("_headers", BindingFlags.Instance | BindingFlags.NonPublic) ??
+            typeof(HttpRequestMessage).GetField("headers", BindingFlags.Instance | BindingFlags.NonPublic);
+
         public static bool HasHeaders(this HttpRequestMessage request)
         {
-            // Note: The field name is _headers in .NET core 
-            bool isDotNetFramework = RuntimeUtils.IsDotNetFramework();
-            bool isDotNetFrameworkOrMono = isDotNetFramework || RuntimeUtils.IsMono();

```

Comment:
So we can remove the IsMono method as well

Reply:
We can remove the whole RuntimeInformation class that is not used anymore. Should I do it in this PR?

Reply:
RuntimeUtils I mean

Reply:
Please remove just the IsMono method